mungegithub/mungers/sig-mention-handler_test.go

```diff
@@ -0,0 +1,162 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	githubapi "github.com/google/go-github/github"
+	"k8s.io/test-infra/mungegithub/github"
+	github_test "k8s.io/test-infra/mungegithub/github/testing"
+	"net/http"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+const (
+	helpWanted    = "help-wanted"
+	open          = "open"
+	sigApps       = "sig/apps"
+	username      = "Ali"
+	needsSigLabel = "needs-sig"
+)
+
+func TestSigMentionHandler(t *testing.T) {
+	runtime.GOMAXPROCS(runtime.NumCPU())
+
+	tests := []struct {
+		name     string
+		issue    *githubapi.Issue
+		expected []githubapi.Label
+	}{
+		{
+			name: "ignore PRs",
+			issue: &githubapi.Issue{
+				State: githubapi.String(open),
+				Labels: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+					githubapi.Label{Name: githubapi.String(sigApps)}},
+				PullRequestLinks: &githubapi.PullRequestLinks{},
+				Assignee:         &githubapi.User{Login: githubapi.String(username)},
+				Number:           intPtr(1),
+			},
+			expected: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+				githubapi.Label{Name: githubapi.String(sigApps)}},
+		},
+		{
+			name:     "issue is null",
+			issue:    nil,
+			expected: nil,
+		},
+		{
+			name: "issue state is null",
+			issue: &githubapi.Issue{
+				State: nil,
+				Labels: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+					githubapi.Label{Name: githubapi.String(sigApps)}},
+				PullRequestLinks: nil,
+				Assignee:         &githubapi.User{Login: githubapi.String(username)},
+				Number:           intPtr(1),
+			},
+			expected: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+				githubapi.Label{Name: githubapi.String(sigApps)}},
+		},
+		{
+			name: "issue is closed",
+			issue: &githubapi.Issue{
+				State: githubapi.String("closed"),
+				Labels: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+					githubapi.Label{Name: githubapi.String(sigApps)}},
+				PullRequestLinks: nil,
+				Assignee:         &githubapi.User{Login: githubapi.String(username)},
+				Number:           intPtr(1),
+			},
+			expected: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+				githubapi.Label{Name: githubapi.String(sigApps)}},
+		},
+		{
+			name: "issue has at least one sig label",
+			issue: &githubapi.Issue{
+				State: githubapi.String(open),
+				Labels: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+					githubapi.Label{Name: githubapi.String(sigApps)}},
+				PullRequestLinks: nil,
+				Assignee:         &githubapi.User{Login: githubapi.String(username)},
+				Number:           intPtr(1),
+			},
+			expected: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+				githubapi.Label{Name: githubapi.String(sigApps)}},
+		},
+		{
+			name: "issue has no sig label",
```

Comment:
Also interesting tests:
* Do not leave a comment if we cannot add the needsSigLabel
* Do not leave a comment if it already has the needsSigLabel